sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/jdbc/JDBCTableCatalog.scala

```diff
@@ -297,4 +300,21 @@ class JDBCTableCatalog extends TableCatalog with SupportsNamespaces with Logging
   private def getTableName(ident: Identifier): String = {
     (ident.namespace() :+ ident.name()).map(dialect.quoteIdentifier).mkString(".")
   }
+
+  override def listFunctions(namespace: Array[String]): Array[Identifier] = {
+    if (namespace.isEmpty || namespaceExists(namespace)) {
```

Comment:
hmm, do we really need to allow dialect to register functions with different namespaces? Maybe we can require the dialect to return `Seq[(String, UnboundFunction)]`, which means the function are always in the empty namspace, and end-users need to call functions with catalog name and function name, e.g. `SELECT h2.func1(...)`

Reply:
OK